hr_timesheet_purchase_order/models/res_partner.py

```diff
@@ -0,0 +1,252 @@
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    is_auto_po_generate = fields.Boolean(
+        string="Automatic PO generation from timesheet sheets"
+    )
+    recurrence_id = fields.Many2one("hr.timesheet.recurrence", copy=False)
+    employee_ids = fields.One2many(
+        comodel_name="hr.employee",
+        inverse_name="billing_partner_id",
+    )
+
+    is_send_po = fields.Boolean(string="Send RFQ by email after creation")
+
+    next_recurrence_date = fields.Date(related="recurrence_id.next_recurrence_date")
+    repeat_interval = fields.Integer(
+        string="Repeat Every", default=1, compute="_compute_repeat", readonly=False
+    )
+    repeat_unit = fields.Selection(
+        [
+            ("day", "Days"),
+            ("week", "Weeks"),
+            ("month", "Months"),
+            ("year", "Years"),
+        ],
+        default="week",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+    repeat_type = fields.Selection(
+        [
+            ("forever", "Forever"),
+            ("until", "End Date"),
+            ("after", "Number of Repetitions"),
+        ],
+        default="forever",
+        string="Until",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+    repeat_until = fields.Date(
+        string="End Date", compute="_compute_repeat", readonly=False
+    )
+    repeat_number = fields.Integer(
+        string="Repetitions", default=1, compute="_compute_repeat", readonly=False
+    )
+
+    repeat_on_month = fields.Selection(
+        [
+            ("date", "Date of the Month"),
+            ("day", "Day of the Month"),
+        ],
+        default="date",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+
+    repeat_on_year = fields.Selection(
+        [
+            ("date", "Date of the Year"),
+            ("day", "Day of the Year"),
+        ],
+        default="date",
+        compute="_compute_repeat",
+        readonly=False,
+    )
+
+    mon = fields.Boolean(string="Mon", compute="_compute_repeat", readonly=False)
+    tue = fields.Boolean(string="Tue", compute="_compute_repeat", readonly=False)
+    wed = fields.Boolean(string="Wed", compute="_compute_repeat", readonly=False)
+    thu = fields.Boolean(string="Thu", compute="_compute_repeat", readonly=False)
+    fri = fields.Boolean(string="Fri", compute="_compute_repeat", readonly=False)
+    sat = fields.Boolean(string="Sat", compute="_compute_repeat", readonly=False)
+    sun = fields.Boolean(string="Sun", compute="_compute_repeat", readonly=False)
+
+    repeat_day = fields.Integer(
+        compute="_compute_repeat",
+        readonly=False,
+    )
+
+    @api.onchange("repeat_day", "repeat_month")
+    def _onchange_repeat_day(self):
+        if 0 > self.repeat_day or self.repeat_day > 31:
+            self.repeat_day = 1
+        if self.repeat_month == "february" and self.repeat_day > 29:
+            self.repeat_day = 28
+
+    repeat_week = fields.Selection(
```

Comment:
The same selection in prev model, would you create variable for this selection and use it in fields?

Reply:
I can't do this because all these fields are calculated or modified by the user